Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -144,21 +196,41 @@ void HiresTexture::Prefetch()
 {
   Common::SetCurrentThreadName("Prefetcher");
 
-  size_t size_sum = 0;
-  size_t sys_mem = Common::MemPhysical();
-  size_t recommended_min_mem = 2 * size_t(1024 * 1024 * 1024);
-  // keep 2GB memory for system stability if system RAM is 4GB+ - use half of memory in other cases
-  size_t max_mem =
-      (sys_mem / 2 < recommended_min_mem) ? (sys_mem / 2) : (sys_mem - recommended_min_mem);
-  u32 starttime = Common::Timer::GetTimeMs();
-  for (const auto& entry : s_textureMap)
+  u32 prefetch_starttime_total = 0;
+  u32 prefetch_stoptime_total = 0;
+
+  // Default 30% -- Todo User Configurable GFX_HIRESTEX_OVERRIDE_PREFETCH_MEM_LIMIT
+  u32 percent_mem_to_cull = 30;
+
+  size_t prefetched_size_sum = 0;  // total mem size incl. non-native decompressed textures
+  size_t absolute_max_mem = Common::MemPhysical();  // Currently "total usable", excl. hw reserved
+  size_t culled_percent_amount = (Common::MemPhysical() * percent_mem_to_cull) / 100;
+  size_t delegated_max_mem = Common::MemPhysical() - culled_percent_amount;
+  size_t below_min_req_mem = 3221225472;  // WIP - 3 GB; 2GB Rec. + 1GB for OS/App/Misc?
```

Comment:
`1024 * 1024 * 1024 * 3`, and I think there's already constants for this in a header file somewhere.

Also all of these should maybe be `const` or `constexpr`, but that one's a judgement call.